pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1348,25 +1347,46 @@ char **NpyArr_encodeLabels(PyArrayObject *labels, PyObjectEncoder *enc,
                     len = strlen(cLabel);
                 }
             }
-        } else {  // Fallback to string representation
-            // Replace item with the string to keep it alive.
-            Py_SETREF(item, PyObject_Str(item));
-            if (item == NULL) {
-                NpyArr_freeLabels(ret, num);
-                ret = 0;
-                break;
+        } else {
+            // NA values need special handling
+            if (PyFloat_Check(item)) {
+                double fval = PyFloat_AS_DOUBLE(item);
+                is_null = npy_isnan(fval);
+            } else if (item == Py_None || object_is_na_type(item)) {
```

Comment:
Do we have a similar expression already in missing.pyx? I wonder if there's a way for us to be consistent about our missing object evaluation

Reply:
``checknull`` is probably the equivalent. Without having a C API, I don't really think we can unify anything sadly.

Reply:
Gotcha. Not a blocker here, but it might be nice to create a separate missing.c file with the proper implementation and then can wrap that from missing.pyx

Reply:
On a second read I think we should at least create a separate function for `int is_null(PyObject *obj)` here. Can just be static in the current translation unit - makes the code more readable than the nested if statements